grr/lib/data_stores/http_data_store.py

```diff
@@ -83,7 +83,10 @@ def _ReadExactly(self, n):
     ret = ""
     left = n
     while left:
-      ret += self.sock.recv(left)
+      data = self.sock.recv(left)
+      if data == "":
+        raise IOError("Expected %d bytes, got EOF after %d", n, len(ret))
```

Comment:
I agree that catching the exception would be better but I also like IOError better than socket error - the socket probably works just fine. Can you maybe add IOError to the exceptions we catch in _Reconnect and _ReadReply?